components/add_users_to_group_modal/add_users_to_group_modal.tsx

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {localizeMessage} from 'utils/utils.jsx';
+import {Group} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import {ModalData} from 'types/actions';
+import AddUserToGroupMultiSelect from 'components/add_user_to_group_multiselect';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+export type Props = {
+    onExited: () => void;
+    groupId: string;
+    group: Group;
+    backButtonCallback: () => void;
+    actions: {
+        addUsersToGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+type State = {
+    show: boolean;
+    savingEnabled: boolean;
+    usersToAdd: UserProfile[];
+    showUnknownError: boolean;
+}
+
+export default class AddUsersToGroupModal extends React.PureComponent<Props, State> {
```

Comment:
nit: We are heading more usage of functional components for components that has less logic, probably this is a good candidate for a functional component.

Reply:
Agreed, changed this to a functional component